src/config.cc

```diff
@@ -5,200 +5,199 @@
-						oss << possible_values[i].second << (i < possible_values.size() - 1 ? ", " : ".");
-					}
-					CELERITY_WARN(oss.str());
-				}
-			}
-
 			// Set both the global log level and the default sink level so that the console logger adheres to CELERITY_LOG_LEVEL even if we temporarily
 			// override the global level in test_utils::log_capture.
 			// TODO do not modify global state in the constructor, but factor the LOG_LEVEL part out of detail::config entirely.
 			spdlog::set_level(log_lvl);
 			for(auto& sink : spdlog::default_logger_raw()->sinks()) {
 				sink->set_level(log_lvl);
 			}
-		}
 
-		// ------------------------- CELERITY_GRAPH_PRINT_MAX_VERTS ---------------------------
+			// ------------------------- CELERITY_GRAPH_PRINT_MAX_VERTS ---------------------------
 
-		{
-			const auto [is_set, value] = get_env("CELERITY_GRAPH_PRINT_MAX_VERTS");
-			if(is_set) {
-				if(spdlog::should_log(log_level::trace)) {
-					CELERITY_WARN("CELERITY_GRAPH_PRINT_MAX_VERTS: Graphs will only be printed for CELERITY_LOG_LEVEL=trace.");
-				}
-				const auto [is_valid, parsed] = parse_uint(value.c_str());
-				if(is_valid) { m_graph_print_max_verts = parsed; }
-			}
-		}
+			const auto has_gpmv = parsed_and_validated_envs.get(env_gpmv);
+			if(has_gpmv) { m_graph_print_max_verts = *has_gpmv; }
 
-		// --------------------------------- CELERITY_DEVICES ---------------------------------
-
-		{
-			const auto [is_set, value] = get_env("CELERITY_DEVICES");
-			if(is_set) {
-				if(value.empty()) {
-					CELERITY_WARN("CELERITY_DEVICES is set but empty - ignoring");
-				} else {
-					std::istringstream ss{value};
-					std::vector<std::string> values{std::istream_iterator<std::string>{ss}, std::istream_iterator<std::string>{}};
-					if(static_cast<long>(m_host_cfg.local_rank) > static_cast<long>(values.size()) - 2) {
-						throw std::runtime_error(fmt::format("Process has local rank {}, but CELERITY_DEVICES only includes {} device(s)",
-						    m_host_cfg.local_rank, values.empty() ? 0 : values.size() - 1));
-					}
-
-					if(static_cast<long>(values.size()) - 1 > static_cast<long>(m_host_cfg.node_count)) {
-						CELERITY_WARN("CELERITY_DEVICES contains {} device indices, but only {} worker processes were spawned on this host", values.size() - 1,
-						    m_host_cfg.node_count);
-					}
-
-					const auto pid_parsed = parse_uint(values[0].c_str());
-					const auto did_parsed = parse_uint(values[m_host_cfg.local_rank + 1].c_str());
-					if(!pid_parsed.first || !did_parsed.first) {
-						CELERITY_WARN("CELERITY_DEVICES contains invalid value(s) - will be ignored");
-					} else {
-						m_device_cfg = device_config{pid_parsed.second, did_parsed.second};
-					}
-				}
-			}
-		}
+			// --------------------------------- CELERITY_DEVICES ---------------------------------
 
-		// ----------------------------- CELERITY_PROFILE_KERNEL ------------------------------
-		{
-			const auto result = get_env("CELERITY_PROFILE_OCL");
```

Comment:
Why has this been dropped completely? It should either still exist and be parsed, but print the warning that it's been renamed, or it should just fail saying that it has been renamed in the error message.